src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,53 @@
+package roomescape.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import roomescape.dao.RoomDAO;
+import roomescape.domain.Reservation;
+import roomescape.domain.Time;
+import roomescape.dto.ReservationDto;
+import roomescape.dto.TimeResDto;
+
+@Service
+public class ReservationService {
+    private final RoomDAO roomDAO;
+    private final TimeService timeService;
+
+    public ReservationService(RoomDAO roomDAO, TimeService timeService) {
+        this.roomDAO = roomDAO;
+        this.timeService = timeService;
+    }
+
+    @Transactional(readOnly = true)
+    public List<ReservationDto> getAllReservations() {
+        return roomDAO.findAll().stream()
+                .map(reservation -> new ReservationDto(reservation.getName(), reservation.getDate(),
+                        reservation.getTime()))
+                .collect(Collectors.toList());
+    }
```

Comment:
```suggestion
    @Transactional(readOnly = true)
    public List<ReservationDto> getAllReservations() {
        return roomDAO.findAll().stream()
                .map(reservation -> new ReservationDto(reservation.getName(), reservation.getDate(),
                        reservation.getTime()))
                .toList();
    }
```

- `collect(Collectors.toList())` vs `toList()` 중에서 전자를 선택하신 이유가 있나요?

Reply:
Collectors.toList()는 가변 리스트를 반환하지만 toList()는 불변 리스트를 반환합니다. 컬렉션을 수정할 필요가 있는 경우 Collectors를 사용하고 불변성을 유지하고 싶을 경우 toList()를 사용한다고 합니다. 결과값을 불변성으로 보장하기 위해 toList()를 사용하는 것이 좋아보이는 것 같습니다 감사합니다! 😊 `collect(Collectors.toList())` 와 `toList()`의 차이를 사실 잘 모르고 사용했는데 이번 기회에 공부하면서 제대로 알게 된 것 같습니다!